### PR TITLE
Add git user profile functions for prompt display

### PR DESCRIPTION
**Pull request by aesopwolf**
Add functions for `git config user.name` and `git config user.email`.

Useful as visual indicator for people who need to switch between different work and home user accounts often.

### CODE REVIEW

**Comment by mcornella** [comment]
You could team up with @dottorblaster over at #4673 and make a dedicated plugin for git user profiles.

**Comment by aesopwolf** [comment]
@mcornella good idea! It's nice to see someone else with a similar problem.

My original plan was to:

1. Get this pull request landed and get my team at work using it
1. Then later I can build a git user switcher via [git-extras](https://github.com/tj/git-extras)

Unless you think a git user switcher should be built as an oh-my-zsh plugin?

**Comment by mcornella** [comment]
I like `git-extras` very much, I use it all the time, so I don't have a problem with building it over there. Having these prompt functions in the git plugin is a little bit weird though, but I guess it's the best option we have. Let me quickly review the functions and we'll get this merged.

File: lib/git.zsh [hunk 1]
```diff
@@ -188,6 +188,24 @@ function git_compare_version() {
   echo 0
 }
 
+# Outputs the name of the current user
+# Usage example: $(git_current_user_name)
+function git_current_user_name() {
+  local ref
+  ref=$(command git config user.name 2> /dev/null)
+  local ret=$?
```
Comment (mcornella) [inline lib/git.zsh]: This `$ret` variable is not used anywhere else, it can be safely deleted.

File: lib/git.zsh [hunk 2]
```diff
@@ -188,6 +188,24 @@ function git_compare_version() {
...
+  echo ${ref}
```
Comment (mcornella) [inline lib/git.zsh]: If you're gonna just `echo` the variable, you can make the whole function be:
```sh
function .. {
  command git config user.name 2>/dev/null
}
```

**Comment by aesopwolf** [comment]
@mcornella thanks for your comments on the code. As you can tell I didn't know exactly what I was doing and had just copy/pasted from the `git_current_branch` function

Anyways, I just updated the two functions to use the one-liner you recommended. Should be good to go now

**Comment by mcornella** [comment]
That's better. Now one last thing, squash the current 4 commits into a single one, and then edit the commit message to be more descriptive (you can use this same PR title). 

If you know how to use rebase you'll know how to do it; otherwise, ask and I'll guide you through the steps.

**Comment by aesopwolf** [comment]
@mcornella oh yeah good call on the rebase. That history was pretty ugly. Would you mind taking another look?

**Comment by mcornella** [comment]
Perfect!

**Comment by aesopwolf** [comment]
Sweeeeeet! Thanks so much :)

**Pull request merged**